File: Lexor/UI/lexor_desktop/lib/models/account_response.dart
class AccountResponse {
  final int id;
  final String firstName;
  final String lastName;
  final String username;
  final String email;
  final String? phoneNumber;
  final String? profileImageBase64;

  AccountResponse({
    required this.id,
    required this.firstName,
    required this.lastName,
    required this.username,
    required this.email,
    this.phoneNumber,
    this.profileImageBase64,
  });

  String get fullName => '$firstName $lastName';

  factory AccountResponse.fromJson(Map<String, dynamic> json) => AccountResponse(
    id: json['id'] as int,
    firstName: json['firstName'] as String? ?? '',
    lastName: json['lastName'] as String? ?? '',
    username: json['username'] as String? ?? '',
    email: json['email'] as String? ?? '',
    phoneNumber: json['phoneNumber'] as String?,
    profileImageBase64: json['profileImageBase64'] as String?,
  );
}

File: Lexor/Lexor.Model/Responses/AccountResponse.cs
namespace Lexor.Model.Responses
{
    // Current user's account (works for both admin and employee).
    public class AccountResponse
    {
        public int Id { get; set; }
        public string FirstName { get; set; } = string.Empty;
        public string LastName { get; set; } = string.Empty;
        public string Username { get; set; } = string.Empty;
        public string Email { get; set; } = string.Empty;
        public string? PhoneNumber { get; set; }
        public string? ProfileImageBase64 { get; set; }
    }
}
